cpp/benchmarks/synchronization/synchronization.hpp

```diff
@@ -0,0 +1,34 @@
+// Google Benchmark library
+#include <benchmark/benchmark.h>
+
+#include <tests/utilities/cudf_test_utils.cuh>
+
+class cuda_event_timer {
+ 
+public:
+  
+  // The two c'tor clear the L2$ by cudaMemset a buffer of L2$ size
+  // and start the timer. 
+  cuda_event_timer(benchmark::State& state);  
+  
+  cuda_event_timer();  
+  
+  // The d'tor stops the timer and perform a synchroniazation. If 
+  // a benchmark::State object is provided to the c'tor its time 
+  // will be set to the value given `cudaEventElapsedTime`.
+  ~cuda_event_timer();
+
+private:
+ 
+  void init();
+
+  cudaEvent_t start, stop;
+  
+  static int l2_cache_bytes;
```

Comment:
What's the thought behind making these `static`?

Reply:
The thought is we probably don't need to make multiple queries about L2$ size in a run? This should be cheap enough though.

Reply:
@jrhemstad But if we are enabling different streams the static buffer does not make sense anymore.